src/libmongoc/tests/test-mongoc-sample-commands.c

```diff
@@ -3018,6 +3056,11 @@ test_sample_causal_consistency (mongoc_client_t *client)
       return;
    }
 
+   if (test_framework_is_replset () && test_framework_data_nodes_count () == 1) {
```

Comment:
Should this condition be `!test_framework_is_replset () || test_framework_data_nodes_count () == 1)`? Does this test support running against a sharded cluster?